test/system/src/test/java/io/pravega/test/system/KvsWithFailOverTest.java

```diff
@@ -0,0 +1,140 @@
+
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+@RunWith(SystemTestRunner.class)
+public class KvsWithFailOverTest extends AbstractFailoverTests {
+    static final String SCOPE_NAME = "kvsScope" + RandomFactory.create().nextInt(Integer.MAX_VALUE);
+    static final String KVT_NAME = "testKvt";
+    static final String KEY_FAMILY = "testKeyFamily";
+    private static final int NUM_KVPWRITE = 5;
+    private static final int NUM_KVPREAD = 5;
+    @Rule
+    //The execution time for @Before + @After + @Test methods should be less than 25 mins. Else the test will timeout.
+    public Timeout globalTimeout = Timeout.seconds(25 * 60);
+    private StreamManager streamManager;
+    private ConnectionFactory connectionFactory;
+    private ConnectionPool connectionPool;
+    private KeyValueTableFactory keyValueTableFactory;
+
+    @Environment
+    public static void initialize() throws MarathonException, ExecutionException {
+        URI zkUri = startZookeeperInstance();
+        startBookkeeperInstances(zkUri);
+    }
+
+    @Before
+    public void setup() throws ExecutionException {
+        // Get zk details to verify if controller, SSS are running
+        Service zkService = Utils.createZookeeperService();
+        List<URI> zkUris = zkService.getServiceDetails();
+        log.debug("Zookeeper service details: {}", zkUris);
+        //get the zk ip details and pass it to  host, controller
+        URI zkUri = zkUris.get(0);
+        // Verify controller is running.
+        controllerInstance = Utils.createPravegaControllerService(zkUri);
+        assertTrue(controllerInstance.isRunning());
+        List<URI> conURIs = controllerInstance.getServiceDetails();
+        log.info("Pravega Controller service instance details: {}", conURIs);
+
+        // Fetch all the RPC endpoints and construct the client URIs.
+        final List<String> uris = conURIs.stream().filter(ISGRPC).map(URI::getAuthority).collect(Collectors.toList());
+        controllerURIDirect = URI.create("tcp://" + String.join(",", uris));
+        log.info("Controller Service direct URI: {}", controllerURIDirect);
+        // Verify segment store is running.
+        segmentStoreInstance = Utils.createPravegaSegmentStoreService(zkUri, controllerURIDirect);
+        assertTrue(segmentStoreInstance.isRunning());
+        log.info("Pravega Segmentstore service instance details: {}", segmentStoreInstance.getServiceDetails());
+        executorService = ExecutorServiceHelpers.newScheduledThreadPool(NUM_KVPWRITE + NUM_KVPREAD + 1, "KvsFailOverTest-main");
+        controllerExecutorService = ExecutorServiceHelpers.newScheduledThreadPool(2, "KvsFailOverTest-controller");
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURIDirect);
+        log.info("BEFORE 1");
+        // Making controller and segmentstore instance count to 1
+        //log.info("controller size {}", (controllerInstance.getServiceDetails().size()) / 2);
```

Comment:
Please delete all commented out code from this PR.

Reply:
Resolved